File: code/error-demo.go
package main

import (
	"errors"
	"fmt"
)

func div(a, b int) (int, error) {
	if b == 0 {
		return 0, errors.New("division by zero")
	}
	return a / b, nil
}
func main() {
	//fmt.Println("Hello, World!")
	result, err := div(10, 2)
	if err != nil {
		fmt.Println("Error:", err)
	} else {
		fmt.Println("Result:", result)
	}
}

File: code/error-demo.js
function div(a,b)
{
    if(b==0)
    {
        return [undefined, new Error("Division by zero error")];
    }
    return [a/b, undefined];
}
let [result, error] = div(10,2);
if(error)
{
    console.log(error.message);
    //let a = 1 + 2;
}
else
{
    console.log(result);
}